iXBRLViewerPlugin/__init__.py

```diff
@@ -81,6 +81,7 @@ def iXBRLViewerCommandLineOptionExtender(parser, *args, **kwargs):
                       default=False,
                       dest="zipViewerOutput",
                       help="Converts the viewer output into a self contained zip")
+    parser.add_option("--keepOpen", dest="keepOpen", default=True, action="store_true", help=argparse.SUPPRESS)
```

Comment:
This is an existing Arelle option. This line just forces it to `True`.  Have added a comment in 169b027

Reply:
Ah, I see.  In that case, when do things get "closed" (e.g., for long-running inside the GUI)?

Reply:
This option only affects the command line.  I'm not actually sure how to open multiple documents simultaneously in the GUI, but I assume the models are held open until the relevant window is closed.

Reply:
Turns out that this dubious approach to overriding the default value doesn't work with the most recent Arelle (interferes with new code that differentiates plugin options from base options).  Changed to a different, but similarly dubious approach in 74b457d8dbfca2392a4d42043c4ca83f1d7b5fdf.

Reply:
Is the new approach backwards compatible, or does it raise the minimum?

@austinmatherne-wk Do you have any ideas on how we could add a more robust solution to Arelle?

Reply:
Should be backwards compatible.  Newer Arelle relies on counting the number of arguments added by the plugin, and gets confused by re-specifying an existing argument.

Reply:
We need to address a breaking change involving the way CLI options work in Arelle. The current Arelle CLI leverages the deprecated [optparse](https://docs.python.org/3/library/optparse.html) library. Our aim is to transition to [argparse](https://docs.python.org/3/library/argparse.html). However, this migration is complicated since plugins are directly importing and using optparse to register arguments.

I’m considering the introduction of a new plugin hook named `CntlrCmdLine.Arguments`. This would allow plugins to return a list of argument objects (based on a new class in Arelle). These arguments would then be applied by the cntrl to configure optparse, argparse, or any command-line library Arelle adopts in the future. For compatibility, I'm exploring ways to delay making this a breaking change, possibly by passing an empty optparse parser to the existing hook, then iterating over it to discover plugin defined arguments and applying them to the new parser. If a user enables a plugin that still uses the old system, `CntlrCmdLine.Options` would log a deprecated plugin hook warning.

I hadn't initially accounted for plugins altering argument options from base Arelle. Based on my plan, the missing `keepOpen` argument would break the implementation in this PR with future versions of Arelle. We might be able to allow the new hook to also return a list of argument option overrides, but I'd suggest limiting changes only to the default value. The help text could then automatically be updated by the cntrl, letting users know which plugin modified the default value. Thoughts?

Reply:
I think that plugins manipulating command line options in this way is fundamentally the wrong approach.  We also do a similar thing with `logToBuffer` but the difference there is that the Arelle core doesn't define that argument, it just reads from it in case a plugin has.

I'm not really sure why `logToBuffer` or `keepOpen` need to be exposed as end user command line options at all.  They seem like internal options on the XBRL processor which plugins rather than end users need to be able to configure, so perhaps we need an alternative way to request these processor options at start up, and then prevent the user from overriding them with command line options?

It's also worth saying that I only used `keepOpen` because I couldn't see a good way to pass context between calls to `Xbrl.Run` plugin callbacks.  If we had that, we could build the viewer incrementally, and wouldn't need `keepOpen`.

Likewise for `logToBuffer`, we use this to extract validation messages for the report.  It would be much nicer if there were an interface for getting validation results directly after loading a model (somewhat related to https://github.com/Arelle/Arelle/issues/760)